packages/devtools/lib/src/timeline/cpu_profile_protocol.dart

```diff
@@ -161,8 +161,11 @@ class CpuStackFrame {
   }
 
   @override
-  String toString() {
-    return '$name ($sampleCount samples, '
-        '${percent2(cpuConsumptionRatio)})';
+  String toString({Duration duration}) {
+    String str = '$name ';
+    if (duration != null) {
+      str += '- ${msText(duration, fractionDigits: 2)} ';
+    }
+    return str += '($sampleCount samples, ${percent2(cpuConsumptionRatio)})';
```

Comment:
nit: return the final concat string at the end
and use StringBuilder for clarity.

Reply:
done.